cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/spark/SparkUtils.java

```diff
@@ -45,16 +47,38 @@
   private static final String SPARK_ASSEMBLY_JAR = "SPARK_ASSEMBLY_JAR";
   // Environment variable name for locating spark home directory
   private static final String SPARK_HOME = "SPARK_HOME";
+  // Environment variable name for locating hadoop and other spark dependencies for hadoop less spark
+  private static final String SPARK_DIST_CLASSPATH = "SPARK_DIST_CLASSPATH";
 
   private static File sparkAssemblyJar;
 
+
+  /**
+   * Locate the spark assembly and its hadoop dependency jars (if hadoop less build) from the local filesystem.
+   *
+   * @return jar location of spark assembly and hadoop dependencies (if hadoop less build)
+   */
+  public static synchronized Set<File> locateSparkDependencyJars() {
+    Set<File> sparkDepdencies = Sets.newHashSet(locateSparkAssemblyJar());
+    // SPARK_DIST_CLASSPATH is set for hadoop-less spark builds and should contain all hadoop dependencies jars
+    // For example CDH deployed through Cloudera Manager has hadoop-less spark
+    String sparkDistClassPath = System.getenv(SPARK_DIST_CLASSPATH);
+    if (sparkDistClassPath != null) {
+      for (String path : sparkDistClassPath.split(File.pathSeparator)) {
+        File file = new File(path).getAbsoluteFile();
+        sparkDepdencies.add(file);
+      }
+    }
+    return sparkDepdencies;
+  }
+
   /**
    * Locates the spark-assembly jar from the local file system.
    *
    * @return the spark-assembly jar location
    * @throws IllegalStateException if cannot locate the spark assembly jar
    */
-  public static synchronized File locateSparkAssemblyJar() {
+  private static synchronized File locateSparkAssemblyJar() {
```

Comment:
You also need to modify the `createSparkFrameworkClassLoader` in this class to use all the spark dependency jars instead of just the assembly jar.